platform/mds/mds/src/main/java/org/motechproject/mds/service/DefaultMotechDataService.java

```diff
@@ -175,6 +179,9 @@ public T retrieve(String primaryKeyName, Object value) {
     @Transactional
     public T update(final T object) {
         validateCredentials(object);
+        if (JDOHelper.getObjectState(object) == ObjectState.TRANSIENT) {
```

Comment:
Will this method of updating allow us to get rid of those doIntransaction from ITs? We already have an updateFromTransient method if I recall